Mongle/Projects/App/Sources/MainView.swift

```diff
@@ -6,29 +6,48 @@
 //  Copyright © 2024 Mongle-iOS. All rights reserved.
 //
 
-import Core
+import CalendarFeature
 import ChatFeature
-import OnBoardingFeature
+import Core
 import HomeFeature
+import OnBoardingFeature
+import ProfileFeature
 import SwiftUI
 
 struct MainView: View {
+    @StateObject var navigationVM = NavigationViewModel(path: [])
     @EnvironmentObject var kakaoAuth: KaKaoAuthCore
-    
+    @State var selectedTabItem = 0
+
     var body: some View {
         if kakaoAuth.customer.accessToken != nil {
-            TabView {
-                HomeView()
-                    .tabItem {
-                        Image(systemName: "house.fill")
-                        Text("Home")
+            NavigationStack(path: $navigationVM.path) {
+                TabView(selection: $selectedTabItem) {
+                    ForEach(TabItem.allCases, id: \.self) { tab in
+                        tab.view
+                            .tabItem {
+                                Image(selectedTabItem == tab.rawValue ? tab.selectedImage : tab.defaultImage)
+                                Text(tab.title)
+                            }
+                            .tag(tab.rawValue)
                     }
-                
-                ChatView()
-                    .tabItem {
-                        Image(systemName: "message.fill")
-                        Text("Chat")
+                } // TabView
+                .navigationDestination(for: Screen.self) { route in
+                    switch route {
+                    case .home:
+                        HomeView()
+                    case .searchCompany:
+                        SearchView()
+                    case .storeDetail:
+                        Text("test")
+                    case .notification:
+                        NotificationView()
                     }
+                } // navigationDestination
+                .environmentObject(navigationVM)
+                .onOpenURL { url in
+                    navigationVM.handleDeeplink(url: url)
+                } // openURL
```

Comment:
이건 개인의 생각에 따라 다를 것 같은데 가독성을 봤을 때 긴 modifier를 맨 밑으로 가면 어떨까요?
저희가 modifier 규칙을 정하면 좋을 것 같아요!
```suggestion
                .environmentObject(navigationVM)
                .onOpenURL { url in
                    navigationVM.handleDeeplink(url: url)
                } // openURL
               .navigationDestination(for: Screen.self) { route in
                    switch route {
                    case .home:
                        HomeView()
                    case .searchCompany:
                        SearchView()
                    case .storeDetail:
                        Text("test")
                    case .notification:
                        NotificationView()
                    }
                } // navigationDestination
```